contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -5325,3 +5336,141 @@ pltsql_remove_current_query_env(void)
 		destroy_failed_transactions_map();
 	} 
 }
+
+/*
+ * Drop statement of babelfish, currently delete extended property as well.
+ */
+static void
+bbf_ExecDropStmt(DropStmt *stmt)
+{
+	int16		db_id;
+	char		*major_name = NULL, *physical_schema_name = NULL;
+	const char	*schema_name = NULL;
+
+	ObjectAddress	address;
+	Relation		relation = NULL;
+	Oid				schema_oid;
+	ListCell		*cell;
+
+	db_id = get_cur_db_id();
+
+	if (stmt->removeType == OBJECT_SCHEMA)
+	{
+		foreach(cell, stmt->objects)
+		{
+			/* Get schema_name */
+			schema_name = get_logical_schema_name(strVal(lfirst(cell)),
+												  true);
+			if (schema_name)
+			{
+				delete_extended_property(2, db_id, schema_name, NULL, NULL, NULL);
+				pfree((char *) schema_name);
+			}
+		}
+	}
+	else if (stmt->removeType == OBJECT_TABLE ||
+			 stmt->removeType == OBJECT_VIEW ||
+			 stmt->removeType == OBJECT_SEQUENCE)
+	{
+		foreach(cell, stmt->objects)
+		{
+			address = get_object_address(stmt->removeType,
+										 lfirst(cell),
+										 &relation,
+										 AccessShareLock,
+										 true);
+
+			if (!relation)
+				return;
+
+			/* Get major_name */
+			major_name = pstrdup(RelationGetRelationName(relation));
+			relation_close(relation, AccessShareLock);
+
+			/* Get schema_name */
+			schema_oid = get_object_namespace(&address);
+			if (OidIsValid(schema_oid))
+			{
+				physical_schema_name = get_namespace_name(schema_oid);
+				if (physical_schema_name)
+				{
+					schema_name = get_logical_schema_name(physical_schema_name,
+														  true);
+					pfree(physical_schema_name);
+				}
+			}
+
+			if (schema_name && major_name)
+			{
+				delete_extended_property(3, db_id, schema_name, major_name,
+										 NULL, NULL);
+			}
+
+			if (schema_name)
+				pfree((char *) schema_name);
+			if (major_name)
+				pfree(major_name);
+		}
+	}
+	else if (stmt->removeType == OBJECT_PROCEDURE ||
+			 stmt->removeType == OBJECT_FUNCTION ||
```

Comment:
how about user defined type?

Reply:
user defined type will be captured by 'stmt->removeType == OBJECT_TYPE'.